### ISSUE TITLE
appMessagesPushUpdate fails quietly on GCD side-thread

### ISSUE THREAD

**Issue by alist** [pebble/pebble-ios-sdk#2]
Handler is never invoked in this circumstance.

Expected behavior would be a not-main-thread exception. 

//thread 1

``` Objc
    __weak HTPebbleDevice * weakSelf = self;
    dispatch_time_t popTime = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(delayInSeconds * NSEC_PER_SEC));
    dispatch_after(popTime, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^(void){
        [weakSelf sendSomething];
    });
```

//Thread 47

``` Objc
-(void)sendSomething{
        NSDictionary *update = @{ @(1):[NSNumber numberWithUint8:1]};
        [self.connectedWatch appMessagesPushUpdate:update onSent:^(PBWatch *watch, NSDictionary *update, NSError *error) {
//this code isn't ever called ever
            if (!error) {
                NSLog(@"Successfully sent");
            }
            else {
                NSLog(@"Error sending : %@", error);
            }
        }];
}
```

**Comment by drodriguez** [comment]
As the documentation for `appMessagesPushUpdate:onSent:` says, the method “Must be called from the main thread”. Calling it from other thread might cause crashes and undefined behaviour.

Can you try invoking the message from the main thread and see if you can reproduce the issue?

Thanks.

PD: We know that forcing a lot of the framework to be invoked from the main thread is restricting and a bad practice. We have to live with that decision in our app too. It might change in the future, but don’t expect it soon.

**Comment by alist** [comment]
This issue is about making it not fail silently.

Didn't catch that in the documentation. I've already solved the problem, of course. 
Thanks!

**Comment by sarfata** [comment]
@alist We have asserts in the code that should be triggered when you are calling those functions from anything but the main thread. We are not sure why they were not triggered in this case and we will look into this. Let's keep this ticket opened. Thanks! 
(Pebble tracking reference: PBL-16514)

**Comment by mrcljx** [comment]
Fixed in (at least) 3.0.0
